### PR TITLE
DR-1271, DR-1300 helm env variables

### PR DESCRIPTION
**Pull request by snf2ye**
Goal of changes:
* Add variable definitions for perf environment. 
* Support removal of environment specific application.properties settings by moving any needed properties to helm. 

This needs to be merged and applied before merging the associated [jade-data-repo PR.](https://github.com/DataBiosphere/jade-data-repo/pull/574) 

Related documentation - "[How Jade Works](https://docs.google.com/document/d/1xpjrhxGFy3Un5B8NJ-PJhlgYaD24i8HXxiazxkuabeA/edit#heading=h.9qzq41hd7a1g)."

### CODE REVIEW

File: perf/datarepo-api.yaml [hunk 1]
```diff
@@ -10,7 +10,10 @@ env:
   DB_STAIRWAY_URI: jdbc:postgresql://perf-jade-gcloud-sqlproxy.perf:5432/stairway
   DB_DATAREPO_URI: jdbc:postgresql://perf-jade-gcloud-sqlproxy.perf:5432/datarepo
   IT_JADE_API_URL: https://jade-perf.datarepo-perf.broadinstitute.org
-  DATAREPO_GCS_ALLOWREUSEEXISTINGBUCKETS: true
+  DATAREPO_RESOURCEID: broad-jade-perf
+  SAM_STEWARDSGROUPEMAIL: JadeStewards-perf@dev.test.firecloud.org
```
Comment (snf2ye) [inline perf/datarepo-api.yaml:14]: I'm not totally confident that these belong here. I would think we would want to the resourceId to be specific to a project.

File: dev/dd/datarepo-api.yaml [hunk 2]
```diff
@@ -8,7 +8,8 @@ env:
   SPRING_PROFILES_ACTIVE: google,cloudsql,dev,dd,perftest
   DB_STAIRWAY_USERNAME: drmanager
   DB_STAIRWAY_URI: jdbc:postgresql://dd-jade-gcloud-sqlproxy.dd:5432/stairway-dd
-  DB_DATAREPO_URI: jdbc:postgresql://dd-jade-gcloud-sqlproxy.dd:5432/datarepo-dd 
+  DB_DATAREPO_URI: jdbc:postgresql://dd-jade-gcloud-sqlproxy.dd:5432/datarepo-dd
+  DATAREPO_DNSNAME: jade.datarepo-dev.broadinstitute.org
```
Comment (ddietterich) [inline dev/dd/datarepo-api.yaml:12]: I think there are separate dns names for each namespace aren't there? I think mine should be
`jade-dd.datarepo-dev.broadinstitute.org`

File: fakeprod/datarepo-api.yaml [hunk 3]
```diff
@@ -9,6 +9,13 @@ env:
   DB_STAIRWAY_USERNAME: drmanager
   DB_STAIRWAY_URI: jdbc:postgresql://terra-jade-gcloud-sqlproxy.data-repo:5432/stairway
   DB_DATAREPO_URI: jdbc:postgresql://terra-jade-gcloud-sqlproxy.data-repo:5432/datarepo
+  DATAREPO_RESOURCEID: datarepo-terra
+  DATAREPO_DNSNAME: jade-terra.datarepo-prod.broadinstitute.org
+  DATAREPO_GCS_BUCKET: broad-datarepo-terra-prod
```
Comment (ddietterich) [inline fakeprod/datarepo-api.yaml:14]: Does prod need the bucket? I think we are using real projects and creating new buckets so we should not need that.

File: fakeprod/datarepo-api.yaml [hunk 4]
```diff
@@ -9,6 +9,13 @@ env:
...
+  DB_STAIRWAY_FORCECLEAN: false
+  SAM_BASEPATH: https://sam.dsde-prod.broadinstitute.org
+  SAM_STEWARDSGROUPEMAIL: Stewards@firecloud.org
```
Comment (ddietterich) [inline fakeprod/datarepo-api.yaml:17]: De we have/steward's group specified for the dev environment?

File: dev/dd/datarepo-api.yaml (hunk 2 above)
Comment (snf2ye) [reply dev/dd/datarepo-api.yaml:12]: Oh, very interesting. 
So, my thought was that anything in the "[application-dev.properties](https://github.com/DataBiosphere/jade-data-repo/pull/574/files#diff-60a8da0b7574e136eb40838da2f2846dL1)" would have been applied to any namespace in the dev environment. I think that was a bad assumption. I'm going to test this out on my personal namespace. 

If this is the case, then I don't think we have dnsName defined anywhere for our personal namespaces.

File: fakeprod/datarepo-api.yaml (hunk 4 above)
Comment (snf2ye) [reply fakeprod/datarepo-api.yaml:17]: I think that is a great question, but as I looked into it, I think it opens a can of worms. 
This group is specified for the dev environment in application.properties, along with a lot of other environment specific variables. I think this is definitely something we should tackle, but I think I've already let the scope creep on this ticket too much.  I added another ticket to cover [this](https://broadworkbench.atlassian.net/browse/DR-1301).  Let me know what you think of this plan.

File: dev/dd/datarepo-api.yaml (hunk 2 above)
Comment (snf2ye) [reply dev/dd/datarepo-api.yaml]: I did a test run and confirmed that jade-sh.datarepo-dev.broadinstitute.org works for my dev namespace.  So, I made the switch! 

But, if I'm understanding correctly, we didn't have this set correctly for our person dev namespaces before. So, I'm confused as to why we haven't run into issues before.

File: fakeprod/datarepo-api.yaml (hunk 4 above)
Comment (ddietterich) [reply fakeprod/datarepo-api.yaml:16]: Sounds good to me

**Comment by ddietterich** [comment]
i am curious why the dns path worked before. Maybe @smark88 has a clue?

**Comment by smark88** [comment]
> i am curious why the dns path worked before. Maybe @smark88 has a clue?

No idea I never even knew about this field. we set

https://github.com/broadinstitute/datarepo-helm-definitions/blob/sh-DR-1271-helm-env-variables/integration/integration-1/datarepo-api.yaml#L12

**Comment by smark88** [comment]
For consistency lets fill this in for integration too and fill in all `IT_JADE_API_URL` vals too.

**Comment by snf2ye** [comment]
I made the change sugguested by @smark88  and documented future work in https://broadworkbench.atlassian.net/browse/DR-1303.

**Pull request merged**